src/Microsoft.AspNet.OData.Shared/Formatter/EdmLibHelpers.cs

```diff
@@ -31,6 +32,8 @@ internal static class EdmLibHelpers
     {
         private static readonly EdmCoreModel _coreModel = EdmCoreModel.Instance;
 
+        private static readonly ConcurrentDictionary<IEdmModel, ConcurrentDictionary<Type, IEdmType>> _cache = new ConcurrentDictionary<IEdmModel, ConcurrentDictionary<Type, IEdmType>>();
```

Comment:
>ConcurrentDictionary<IEdmModel, ConcurrentDictionary<Type, IEdmType>> [](http://example.com/codeflow?start=32&length=69)

We already have a Cache for the mapping between CLR Type and Edm Type. So, I don't understand why do we need another cache.

You can find cache :

[https://github.com/OData/WebApi/blob/master/src/Microsoft.AspNet.OData.Shared/Formatter/ClrTypeCache.cs](https://github.com/OData/WebApi/blob/master/src/Microsoft.AspNet.OData.Shared/Formatter/ClrTypeCache.cs)

Would you please share me more usage about the GetEdmType(...)?

And By the way, it's recommended to use 
[https://github.com/OData/WebApi/blob/master/src/Microsoft.AspNet.OData.Shared/Formatter/ClrTypeCache.cs#L22](https://github.com/OData/WebApi/blob/master/src/Microsoft.AspNet.OData.Shared/Formatter/ClrTypeCache.cs#L22)

and 
[https://github.com/OData/WebApi/blob/master/src/Microsoft.AspNet.OData.Shared/Formatter/ClrTypeCache.cs#L34](https://github.com/OData/WebApi/blob/master/src/Microsoft.AspNet.OData.Shared/Formatter/ClrTypeCache.cs#L34)

to get CLR type or Edm Type.

Reply:
Sorry I am not very familiar with OData library. While looking at my service which is hosted using OData library I noticed that the GetEdmType was taking up 12% of our CPU.

![image](https://user-images.githubusercontent.com/88375451/127997966-5550fcdd-d511-4957-a68c-0ab0ebe2d31d.png)


I see that the version we are using does not have the cache the new version of the code is seems to be doing the right caching. Thanks for pointing it out.